app/logic/search.py

```diff
@@ -0,0 +1,40 @@
+from app.models.student import Student
+from app.models.laborStatusForm import LaborStatusForm
+from app.models.department import Department
+from app.models.formHistory import FormHistory
+
+def limitSearch(students, currentUser):
+    """ Given a list of student dictionaries and the currentUser the function will only return a
+    list of students who have held a labor position within the currentUsers department."""
+    newstudents = []
+    # Query returns a list of all departments the currentUser has either been a supervisor or created an LSF for.
+    departments = list(Department.select(Department.departmentID)
+                    .join_from(Department, LaborStatusForm)
+                    .join_from(LaborStatusForm, FormHistory)
+                    .where((LaborStatusForm.supervisor == currentUser.supervisor.ID) | (FormHistory.createdBy == currentUser))
+                    .distinct()
+                    )

```

Comment:
This query is essentially the same as the new one in form search. Let's bring it out into its own function so we can a) test it and b) re-use it.